wakaama.cmake

```diff
@@ -27,52 +27,72 @@ option(WAKAAMA_DATA_OLD_CONTENT_FORMAT "Support the deprecated content format va
 # CoAP
 option(WAKAAMA_COAP_RAW_BLOCK1_REQUESTS "Pass each unprocessed block 1 payload to the application" OFF)
 
-set(WAKAAMA_COAP_DEFAULT_BLOCK_SIZE
-    1024
-    CACHE STRING "Default CoAP block size for block-wise transfers"
-)
-set_property(
-    CACHE WAKAAMA_COAP_DEFAULT_BLOCK_SIZE
-    PROPERTY STRINGS
-             LOG_DISABLED
-             16
-             32
-             64
-             128
-             256
-             512
-             1024
-)
+if(NOT WAKAAMA_COAP_DEFAULT_BLOCK_SIZE)
```

Comment:
Not sure what exactly the issue was before, but set() has been greatly improved [starting with CMake 3.21](https://cmake.org/cmake/help/latest/policy/CMP0126.html#policy:CMP0126). I suggest we try updating cmake_minimum_required() to (at least) CMake 3.21 and check this again.